packages/auto-id/src/utils.ts

```diff
@@ -26,8 +39,278 @@ export function derEncodeSignatureAlgorithmOID(
   return new Uint8Array(derEncoded)
 }
 
-export function addDaysToCurrentDate(days: number): Date {
+export const addDaysToCurrentDate = (days: number): Date => {
   const currentDate = new Date() // This gives you the current date and time
   currentDate.setUTCDate(currentDate.getUTCDate() + days) // Adds the specified number of days
   return currentDate
 }
+
+// Errors from the auto-id pallet.
+export enum AutoIdError {
+  UnknownIssuer = 'UnknownIssuer',
+  UnknownAutoId = 'UnknownAutoId',
+  InvalidCertificate = 'InvalidCertificate',
+  InvalidSignature = 'InvalidSignature',
+  CertificateSerialAlreadyIssued = 'CertificateSerialAlreadyIssued',
+  ExpiredCertificate = 'ExpiredCertificate',
+  CertificateRevoked = 'CertificateRevoked',
+  CertificateAlreadyRevoked = 'CertificateAlreadyRevoked',
+  NonceOverflow = 'NonceOverflow',
+  AutoIdIdentifierAlreadyExists = 'AutoIdIdentifierAlreadyExists',
+  AutoIdIdentifierMismatch = 'AutoIdIdentifierMismatch',
+  PublicKeyMismatch = 'PublicKeyMismatch',
+}
+
+/**
+ * Maps an error code to the corresponding enum variant.
+ * @param errorCode The error code as a hexadecimal string (e.g., "0x09000000").
+ * @returns The corresponding enum variant or null if not found.
+ */
+export const mapErrorCodeToEnum = (errorCode: string): AutoIdError | null => {
+  // Remove the '0x' prefix and extract the relevant part of the error code
+  const relevantPart = errorCode.slice(2, 4).toLowerCase() // Gets the byte corresponding to the specific error
+
+  switch (relevantPart) {
+    case '00':
+      return AutoIdError.UnknownIssuer
+    case '01':
+      return AutoIdError.UnknownAutoId
+    case '02':
+      return AutoIdError.InvalidCertificate
+    case '03':
+      return AutoIdError.InvalidSignature
+    case '04':
+      return AutoIdError.CertificateSerialAlreadyIssued
+    case '05':
+      return AutoIdError.ExpiredCertificate
+    case '06':
+      return AutoIdError.CertificateRevoked
+    case '07':
+      return AutoIdError.CertificateAlreadyRevoked
+    case '08':
+      return AutoIdError.NonceOverflow
+    case '09':
+      return AutoIdError.AutoIdIdentifierAlreadyExists
+    case '0a':
+      return AutoIdError.AutoIdIdentifierMismatch
+    case '0b':
+      return AutoIdError.PublicKeyMismatch
+    default:
+      return null // Or handle unknown error types differently
+  }
+}
+
+export const signAndSendTx = async (
```

Comment:
I think it makes more sense to have this function the `auto-utils` package. It should be generic across extrinsic types, if the [signAndSendTx](https://github.com/subspace/auto-sdk/blob/63d663243dde81ec720c8bb55b4e6c137c5fd40c/packages/auto-utils/src/utils/signAndSendTx.ts#L16) in auto-utils needs to better handle error messages then we should adapt it to do so in a way that is agnostic to the individual extrinsics.

Reply:
Implemented a unified function in auto-utils instead.
https://github.com/subspace/auto-sdk/pull/75/commits/4e5d834d6a415548812313145472a1c694490dc7